conference.js

```diff
@@ -427,25 +429,42 @@ export default {
             };
         }
 
-        let audioAndVideoError, audioOnlyError;
+        let audioAndVideoError,
+            audioOnlyError,
+            tracksCreated;
 
         return JitsiMeetJS.init(config).then(() => {
-            return Promise.all([
+            let tryCreateLocalTracks =
                 // try to retrieve audio and video
                 createLocalTracks(['audio', 'video'])
-                // if failed then try to retrieve only audio
+                    // if failed then try to retrieve only audio
                     .catch(err => {
                         audioAndVideoError = err;
                         return createLocalTracks(['audio']);
                     })
-                // if audio also failed then just return empty array
+                    // if audio also failed then just return empty array
                     .catch(err => {
                         audioOnlyError = err;
                         return [];
-                    }),
+                    })
+                    .then(tracks => {
+                        tracksCreated = true;
+                        return tracks;
+                    });
+
+            window.setTimeout(() => {
```

Comment:
Can we extract this code block in a separate function? We should be very careful when adding more code to the init logic, because it's already very complicated.